qutebrowser/mainwindow/tabwidget.py

```diff
@@ -150,7 +160,22 @@ def update_tab_title(self, idx, field=None):
 
         fields = self.get_tab_fields(idx)
         fields['title'] = fields['title'].replace('&', '&&')
-        fields['index'] = idx + 1
+
+        fields['index'] = str(idx + 1)
+        if config.val.tabs.tree_tabs:
+            # we remove the first two chars because every tab is child of tree
+            # root and that gets rendered as well
+            rendered_tree = self.tree_root.render()
+            tree_prefixes = [pre[2:] for pre, _ in rendered_tree]
```

Comment:
I'm not 100% sure, but can we avoid looping over the rendered tree by only doing the slice on the element we need?